cmd/dockerd/service_windows.go

```diff
@@ -183,6 +184,37 @@ func registerService() error {
 		return err
 	}
 	defer s.Close()
+
+	// See http://stackoverflow.com/questions/35151052/how-do-i-configure-failure-actions-of-a-windows-service-written-in-go
+	const (
+		SC_ACTION_NONE        = 0
+		SC_ACTION_RESTART     = 1
+		SC_ACTION_REBOOT      = 2
+		SC_ACTION_RUN_COMMAND = 3
+
+		SERVICE_CONFIG_FAILURE_ACTIONS = 2
+	)
+
+	type SERVICE_FAILURE_ACTIONS struct {
+		ResetPeriod  uint32
+		RebootMsg    *uint16
+		Command      *uint16
+		ActionsCount uint32
+		Actions      uintptr
+	}
+
+	type SC_ACTION struct {
+		Type  uint32
+		Delay uint32
+	}
+	t := []SC_ACTION{
+		{Type: SC_ACTION_RESTART, Delay: uint32(60000)},
```

Comment:
For clarity, probably better to write these durations as something like:

60 * time.Second / time.Millisecond